NotificationHubs/src/com/windowsazure/messaging/AppleTemplateRegistration.java

```diff
@@ -45,16 +57,22 @@ public String getExpiry() {
 	public void setExpiry(String expiry) {
 		this.expiry = expiry;
 	}
-	
-	
 
+	public Map<String, String> getHeaders() {
+		return headers;
```

Comment:
IMHO, we shouldn't be handing back a reference to this object because it gives the caller a mutable copy. I'd do this:

``` Java
public Map<String, String> getHeaders() {
    return new HashMap<String, String>(headers);
}
```